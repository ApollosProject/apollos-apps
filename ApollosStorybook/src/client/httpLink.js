import { Platform } from 'react-native';
import { createUploadLink } from 'apollo-upload-client';
import ApollosConfig from '@apollosproject/config';
import { split, createHttpLink } from '@apollo/client';
import { getMainDefinition } from '@apollo/client/utilities';

let uri = ApollosConfig.APP_DATA_URL || 'http://localhost:4000/graphql';

export default split(
  ({ query }) => {
    const { kind, operation } = getMainDefinition(query);
    return kind === 'OperationDefinition' && operation === 'mutation';
  },
  createUploadLink({ uri }),
  createHttpLink({
    uri,
    useGETForQueries: true,
  })
);
