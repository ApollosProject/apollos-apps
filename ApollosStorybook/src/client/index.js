import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { ApolloProvider } from 'react-apollo';
import { ApolloProvider as ApolloHookProvider } from '@apollo/react-hooks';
import { ApolloClient } from 'apollo-client';
import { ApolloLink } from 'apollo-link';
import { InMemoryCache } from 'apollo-cache-inmemory';

import httpLink from './httpLink';

const link = ApolloLink.from([httpLink]);
const cache = new InMemoryCache();

export const client = new ApolloClient({
  link,
  cache,
});

// Hack to give auth link access to method on client;
// eslint-disable-next-line prefer-destructuring

class ClientProvider extends PureComponent {
  static propTypes = {
    client: PropTypes.shape({
      cache: PropTypes.shape({}),
    }),
    children: PropTypes.oneOfType([
      PropTypes.arrayOf(PropTypes.node),
      PropTypes.node,
      PropTypes.object, // covers Fragments
    ]).isRequired,
  };

  static defaultProps = {
    client,
  };

  render() {
    const { children, ...otherProps } = this.props;
    return (
      <ApolloProvider {...otherProps} client={client}>
        <ApolloHookProvider {...otherProps} client={client}>
          {children}
        </ApolloHookProvider>
      </ApolloProvider>
    );
  }
}

export default ClientProvider;
