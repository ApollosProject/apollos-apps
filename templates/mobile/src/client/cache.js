import { InMemoryCache } from '@apollo/client/cache';
import AsyncStorage from '@react-native-community/async-storage';
import { CachePersistor } from 'apollo3-cache-persist';
import ApollosConfig from '@apollosproject/config';
import fragmentTypes from '../../fragmentTypes.json';

// We reset our apollo cache based an env value and static number.
// In the future, we should also look at resetting the app when an error occurs related to Apollo.
// You can also increment this number to force a manual reset of the cache.
const SCHEMA_VERSION = `${ApollosConfig.SCHEMA_VERSION}-1`; // Must be a string.
const SCHEMA_VERSION_KEY = 'apollo-schema-version';
const possibleTypes = {};
fragmentTypes.__schema.types.forEach((supertype) => {
  if (supertype.possibleTypes) {
    possibleTypes[supertype.name] = [
      ...supertype.possibleTypes.map((subtype) => subtype.name),
    ];
  }
});

const cache = new InMemoryCache({
  possibleTypes,
  cacheRedirects: {
    Query: {
      fields: {
        node: {
          read(_, { args, toReference }) {
            return toReference({
              __typename: args.id.split(':')[0],
              id: args.id,
            });
          },
        },
      },
    },
  },
});

const persistor = new CachePersistor({
  cache,
  storage: AsyncStorage,
});

// 1. If we are on the same schema version, restore the cache.
// 2. If that fails, purge the cache and update the stored version so we don't try and restore agian
// 3. If we are on a new schema version, purge the cache.
// 4. If purging or setting AsyncStorage keys fails (or anything else) we capture the error and log it.

export const ensureCacheHydration = (async () => {
  // We wrap everything in try/catch because crashing on a cache restore is bad
  // ðŸ˜¬
  try {
    const currentVersion = await AsyncStorage.getItem(SCHEMA_VERSION_KEY);
    if (currentVersion === SCHEMA_VERSION) {
      // If the current version matches the latest version,
      // we're good to go and can restore the cache.
      try {
        await persistor.restore();
      } catch (restoreError) {
        // If the restore fails, we want to do our best to purge the cache.
        await persistor.purge();
        console.error('Error restoring cache, purging the cache', restoreError);
      }
    } else {
      // Otherwise, we'll want to purge the outdated persisted cache
      // and mark ourselves as having updated to the latest version.
      await persistor.purge();
      await AsyncStorage.setItem(SCHEMA_VERSION_KEY, SCHEMA_VERSION);
    }
  } catch (error) {
    console.error('Error restoring or purging Apollo cache', error);
  }
})();

export default cache;
