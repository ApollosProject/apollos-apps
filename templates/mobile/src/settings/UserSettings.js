import { ScrollView } from 'react-native';
import { useQuery, useApolloClient, gql } from '@apollo/client';
import { useNavigation } from '@react-navigation/native';
import { getVersion, getBuildNumber } from 'react-native-device-info';
import { get } from 'lodash';
import Color from 'color';

import {
  BackgroundView,
  TableView,
  Row,
  Divider,
  Touchable,
  NavigationService,
  H3,
  H6,
  PaddedView,
  styled,
} from '@apollosproject/ui-kit';

import {
  UserAvatarUpdate,
  usePersonFollowing,
  usePersonFollowedBy,
  useCurrentUserFollowRequests,
} from '@apollosproject/ui-connected';

import {
  checkOnboardingStatusAndNavigate,
  onboardingComplete,
} from 'onboarding';
import { useLogout } from 'auth';

const Container = styled({
  alignItems: 'center',
  justifyContent: 'center',
})(PaddedView);

const AppVersion = styled(({ theme }) => ({
  color: theme.colors.neutral.gray3,
  textAlign: 'center',
  paddingVertical: theme.sizing.baseUnit,
}))(H6);

const NewLabel = styled(({ theme }) => ({
  color: theme.colors.primary,
  backgroundColor: Color(theme.colors.primary).fade(0.8).string(),
  textAlign: 'center',
  paddingVertical: theme.sizing.baseUnit * 0.25,
  paddingHorizontal: theme.sizing.baseUnit * 0.5,
  borderRadius: 8,
  overflow: 'hidden',
}))(H6);

const UserSettings = () => {
  const navigation = useNavigation();
  const logout = useLogout();
  const client = useApolloClient();
  const { data } = useQuery(gql`
    query currentUserId {
      currentUser {
        id
        profile {
          id
          firstName
          lastName
          campus {
            id
            name
          }
        }
      }

      likedContent {
        totalCount
      }
    }
  `);
  const { total: totalFollowing } = usePersonFollowing(
    data?.currentUser?.profile?.id
  );
  const { total: totalFollowers } = usePersonFollowedBy(
    data?.currentUser?.profile?.id
  );
  const { total: totalRequests } = useCurrentUserFollowRequests();

  const firstName = get(data, 'currentUser.profile.firstName');
  const lastName = get(data, 'currentUser.profile.lastName');
  const campusName = get(data, 'currentUser.profile.campus.name');
  const likedCount = get(data, 'likedContent.totalCount');

  //if (loading) {
  //return <ActivityIndicator />;
  //}
  return (
    <BackgroundView avoidHeader>
      <ScrollView>
        <Container>
          <UserAvatarUpdate />
          <H3>{firstName && lastName ? `${firstName} ${lastName}` : ''}</H3>
        </Container>
        <PaddedView />
        <TableView>
          <Touchable
            onPress={() => {
              navigation.navigate('Following', {
                personId: data?.currentUser?.profile?.id,
                screen: 'followed_by',
              });
            }}
          >
            <Row
              title="Following Me"
              leadingIcon="users-three"
              accessoryText={`${totalFollowers}`}
              accessoryComponent={totalRequests > 0 && <NewLabel>New</NewLabel>}
            />
          </Touchable>

          <Touchable
            onPress={() => {
              navigation.navigate('Following', {
                personId: data?.currentUser?.profile?.id,
                screen: 'following',
              });
            }}
          >
            <Row
              title="I'm Following"
              leadingIcon="user-circle"
              accessoryText={`${totalFollowing}`}
            />
          </Touchable>

          <Touchable
            onPress={() => {
              navigation.navigate('LikedContentFeedConnected');
            }}
          >
            <Row
              title="Liked Content"
              leadingIcon="heart"
              accessoryText={`${likedCount || 0}`}
            />
          </Touchable>
        </TableView>

        <PaddedView />

        <TableView headerText="My Profile">
          <Touchable
            onPress={() => {
              navigation.navigate('PersonalDetails');
            }}
          >
            <Row title="Personal Details" leadingIcon="user" />
          </Touchable>

          <Touchable
            onPress={() => {
              navigation.navigate('Location');
            }}
          >
            <Row
              title="My Campus"
              leadingIcon="navigation-arrow"
              accessoryText={campusName || 'Select'}
            />
          </Touchable>
        </TableView>

        <PaddedView />

        <TableView headerText="Settings">
          <Touchable
            onPress={() => {
              navigation.navigate('ChangePassword');
            }}
          >
            <Row title="Change Password" leadingIcon="lock-simple" />
          </Touchable>

          <Touchable
            onPress={() => {
              navigation.navigate('Notifications');
            }}
          >
            <Row title="Notification Settings" leadingIcon="bell" />
          </Touchable>
        </TableView>

        <PaddedView />

        <TableView>
          <Touchable
            onPress={() => {
              logout();
              // This resets the navigation stack, and the navigates to the first auth screen.
              // This ensures that user isn't navigated to a subscreen of Auth, like the pin entry screen.
              NavigationService.resetToAuth();
            }}
          >
            <Row title="Logout" leadingIcon="arrow-square-out" />
          </Touchable>
        </TableView>

        <AppVersion>
          {`App Version: ${getVersion()}.${getBuildNumber()}`}
        </AppVersion>

        {/* testing panel */}
        {process.env.NODE_ENV !== 'production' ? (
          <>
            <PaddedView />
            <TableView headerText="Developer Tools">
              <Touchable
                onPress={() => NavigationService.resetToAuth('OpenIDConnected')}
              >
                <Row title="Launch OpenID Flow" />
              </Touchable>
              <Divider />
              <Touchable
                onPress={() =>
                  checkOnboardingStatusAndNavigate({
                    latestOnboardingVersion: 2,
                    navigation: NavigationService,
                    client,
                  })
                }
              >
                <Row title="Launch Onboarding" />
              </Touchable>
              <Divider />
              <Touchable
                onPress={() =>
                  onboardingComplete({
                    version: 0,
                    userId: data?.currentUser?.id,
                  })
                }
              >
                <Row title="Reset Onboarding to Unseen" />
              </Touchable>
              <Divider />
              <Touchable
                onPress={() =>
                  onboardingComplete({
                    version: 1,
                    userId: data?.currentUser?.id,
                  })
                }
              >
                <Row title="Reset Onboarding to Seen v1" />
              </Touchable>
            </TableView>
            <PaddedView />
          </>
        ) : null}
      </ScrollView>
    </BackgroundView>
  );
};

export default UserSettings;
