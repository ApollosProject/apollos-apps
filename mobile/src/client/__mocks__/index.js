import { MockedProvider } from 'react-apollo/test-utils';
import ApolloClient from 'apollo-client';
import { makeExecutableSchema, addMockFunctionsToSchema } from 'graphql-tools';
import { SchemaLink } from 'apollo-link-schema';
import cache from '../cache';

import clientStateLink from '../clientStateLink';
import { ApolloLink } from 'apollo-link';

// App refers to '../../../../api/src'
// See config in .babelrc
// eslint-disable-next-line import/no-absolute-path
import { schema as typeDefs } from '/api/data';

export default MockedProvider;

const schema = makeExecutableSchema({ typeDefs });
addMockFunctionsToSchema({ schema });

const link = ApolloLink.from([clientStateLink, new SchemaLink({ schema })]);

export const client = new ApolloClient({
  cache,
  link,
});
