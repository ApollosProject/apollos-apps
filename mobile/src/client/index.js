import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { ApolloProvider } from 'react-apollo';
import { ApolloClient } from 'apollo-client';
import { ApolloLink } from 'apollo-link';

import httpLink from './httpLink';
import clientStateLink from './clientStateLink';
import authLink from './authLink'; // eslint-disable-line
import cache, { ensureCacheHydration } from './cache';

const link = ApolloLink.from([clientStateLink, authLink, httpLink]);

export const client = new ApolloClient({
  link,
  cache,
  queryDeduplication: true,
});

class ClientProvider extends PureComponent {
  static propTypes = {
    client: PropTypes.shape({
      cache: PropTypes.shape({}),
    }),
  };

  static defaultProps = {
    client,
  };

  async componentDidMount() {
    await ensureCacheHydration;
  }

  render() {
    return <ApolloProvider {...this.props} client={client} />;
  }
}

export default ClientProvider;
