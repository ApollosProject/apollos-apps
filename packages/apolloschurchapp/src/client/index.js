import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { ApolloProvider } from 'react-apollo';
import { ApolloClient } from 'apollo-client';
import { ApolloLink } from 'apollo-link';
import SplashScreen from 'react-native-splash-screen';
import gql from 'graphql-tag';

import httpLink from './httpLink';
import clientStateLink from './clientStateLink';
import authLink from './authLink'; // eslint-disable-line
import cache, { ensureCacheHydration } from './cache';

const link = clientStateLink.concat(ApolloLink.from([authLink, httpLink]));

export const client = new ApolloClient({
  link,
  cache,
  queryDeduplication: true,
  shouldBatch: true,
});

client.onResetStore(clientStateLink.writeDefaults);

export const CACHE_LOADED = gql`
  query {
    cacheLoaded @client
  }
`;

export const MARK_CACHE_LOADED = gql`
  mutation markCacheLoaded {
    cacheMarkLoaded @client
  }
`;

class ClientProvider extends PureComponent {
  static propTypes = {
    client: PropTypes.shape({
      cache: PropTypes.shape({}),
    }),
  };

  static defaultProps = {
    client,
  };

  async componentDidMount() {
    try {
      await ensureCacheHydration;
    } catch (e) {
      throw e;
    } finally {
      if (SplashScreen && SplashScreen.hide) SplashScreen.hide();
      client.mutate({ mutation: MARK_CACHE_LOADED });
    }
  }

  render() {
    return <ApolloProvider {...this.props} client={client} />;
  }
}

export default ClientProvider;
