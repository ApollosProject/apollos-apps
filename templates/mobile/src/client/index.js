import { useEffect } from 'react';
import PropTypes from 'prop-types';
import { ApolloProvider, ApolloClient, ApolloLink, gql } from '@apollo/client';
import { createPersistedQueryLink } from '@apollo/client/link/persisted-queries';
import { sha256 } from 'react-native-sha256';
import { getVersion, getApplicationName } from 'react-native-device-info';
import { Platform } from 'react-native';
import { createUploadLink } from 'apollo-upload-client';
import ApollosConfig from '@apollosproject/config';
import AsyncStorage from '@react-native-community/async-storage';

import { authLink, buildErrorLink } from '@apollosproject/ui-authentication';
// import { updatePushId } from '@apollosproject/ui-notifications';

import { NavigationService } from '@apollosproject/ui-kit';

import cache, { ensureCacheHydration } from './cache';

const wipeData = () =>
  cache.writeQuery({
    query: gql`
      query {
        isLoggedIn @client
        cacheLoaded @client
      }
    `,
    data: {
      __typename: 'Query',
      cacheLoaded: false,
      isLoggedIn: false,
    },
  });

let storeIsResetting = false;
const onAuthError = async () => {
  AsyncStorage.removeItem('accessToken');
  AsyncStorage.removeItem('refreshToken');
  if (!storeIsResetting) {
    storeIsResetting = true;
    await client.stop();
    await client.clearStore();
  }
  storeIsResetting = false;
  NavigationService.resetToAuth();
};

// Android's emulator requires localhost network traffic to go through 10.0.2.2
const uri = ApollosConfig.APP_DATA_URL.replace(
  'localhost',
  Platform.OS === 'android' ? '10.0.2.2' : 'localhost'
);

const errorLink = buildErrorLink(onAuthError, uri);
const apqLink = createPersistedQueryLink({
  sha256,
  useGETForHashedQueries: true,
});

const link = ApolloLink.from([
  authLink,
  errorLink,
  apqLink,
  createUploadLink({
    uri,
    headers: {
      // Stops Fetch (and the underlying HTTP stack) from caching on our behalf.
      // Caching should be managed by our server and apollo-client
      // Fetch handles the `cache-control: private` option in a way that
      // causes issues when logging in as a different user.
      'Cache-Control': 'no-cache, no-store',
      ...(ApollosConfig.CHURCH_HEADER
        ? { 'x-church': ApollosConfig.CHURCH_HEADER }
        : {}),
    },
  }),
]);

export const client = new ApolloClient({
  link,
  cache,
  queryDeduplication: false,
  shouldBatch: true,
  name: getApplicationName(),
  version: getVersion(),
  defaultOptions: {
    watchQuery: {
      nextFetchPolicy(lastFetchPolicy) {
        if (
          lastFetchPolicy === 'cache-and-network' ||
          lastFetchPolicy === 'network-only'
        ) {
          return 'cache-first';
        }
        return lastFetchPolicy;
      },
    },
  },
});

wipeData();
// Ensure that media player still works after logout.
client.onClearStore(() => wipeData());

const ClientProvider = ({ children }) => {
  useEffect(() => {
    const initialize = async () => {
      await ensureCacheHydration;
      client.writeQuery({
        query: gql`
          query {
            cacheLoaded @client
          }
        `,
        data: {
          cacheLoaded: true,
        },
      });
    };
    initialize();
  }, []);
  return <ApolloProvider client={client}>{children}</ApolloProvider>;
};

ClientProvider.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
    PropTypes.object, // covers Fragments
  ]).isRequired,
};

export default ClientProvider;
