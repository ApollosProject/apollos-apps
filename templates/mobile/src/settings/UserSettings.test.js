import { create, act } from 'react-test-renderer';
import { gql } from '@apollo/client';
import { MockedProvider } from '@apollo/client/testing';

import UserSettings from './UserSettings';

const mock = {
  request: {
    query: gql`
      query currentUserId {
        currentUser {
          id
        }
      }
    `,
  },
  result: {
    data: {
      currentUser: {
        id: 1,
      },
    },
  },
};

describe('UserSettings component', () => {
  it('renders UserSettings when logged in', async () => {
    let tree;
    act(() => {
      tree = create(
        <MockedProvider mocks={[mock]}>
          <UserSettings />
        </MockedProvider>
      );
    });
    await act(async () => {
      await new Promise((resolve) => setTimeout(resolve, 0));
    });
    expect(tree).toMatchSnapshot();
  });
});
