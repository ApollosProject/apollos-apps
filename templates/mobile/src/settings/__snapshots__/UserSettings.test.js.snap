// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSettings component renders UserSettings when logged in 1`] = `
<BackgroundView
  avoidHeader={true}
>
  <RCTScrollView>
    <View>
      <PaddedView
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <UserAvatarUpdate />
        <H3>
          
        </H3>
      </PaddedView>
      <PaddedView />
      <TableView>
        <Touchable
          onPress={[Function]}
        >
          <Row
            accessoryComponent={
              <Unknown>
                New
              </Unknown>
            }
            accessoryText="1"
            leadingIcon="users-three"
            title="Following Me"
          />
        </Touchable>
        <Touchable
          onPress={[Function]}
        >
          <Row
            accessoryText="1"
            leadingIcon="user-circle"
            title="I'm Following"
          />
        </Touchable>
        <Touchable
          onPress={[Function]}
        >
          <Row
            accessoryText="0"
            leadingIcon="heart"
            title="Liked Content"
          />
        </Touchable>
      </TableView>
      <PaddedView />
      <TableView
        headerText="My Profile"
      >
        <Touchable
          onPress={[Function]}
        >
          <Row
            leadingIcon="user"
            title="Personal Details"
          />
        </Touchable>
        <Touchable
          onPress={[Function]}
        >
          <Row
            accessoryText="Select"
            leadingIcon="navigation-arrow"
            title="My Campus"
          />
        </Touchable>
      </TableView>
      <PaddedView />
      <TableView
        headerText="Settings"
      >
        <Touchable
          onPress={[Function]}
        >
          <Row
            leadingIcon="lock-simple"
            title="Change Password"
          />
        </Touchable>
        <Touchable
          onPress={[Function]}
        >
          <Row
            leadingIcon="bell"
            title="Notification Settings"
          />
        </Touchable>
      </TableView>
      <PaddedView />
      <TableView>
        <Touchable
          onPress={[Function]}
        >
          <Row
            leadingIcon="arrow-square-out"
            title="Logout"
          />
        </Touchable>
      </TableView>
      <H6
        style={[Function]}
      >
        App Version: 1.0.0.1
      </H6>
      <PaddedView />
      <TableView
        headerText="Developer Tools"
      >
        <Touchable
          onPress={[Function]}
        >
          <Row
            title="Launch OpenID Flow"
          />
        </Touchable>
        <Divider />
        <Touchable
          onPress={[Function]}
        >
          <Row
            title="Launch Onboarding"
          />
        </Touchable>
        <Divider />
        <Touchable
          onPress={[Function]}
        >
          <Row
            title="Reset Onboarding to Unseen"
          />
        </Touchable>
        <Divider />
        <Touchable
          onPress={[Function]}
        >
          <Row
            title="Reset Onboarding to Seen v1"
          />
        </Touchable>
      </TableView>
      <PaddedView />
    </View>
  </RCTScrollView>
</BackgroundView>
`;

exports[`UserSettings component renders UserSettings when logged in 2`] = `
<BackgroundView
  avoidHeader={true}
>
  <RCTScrollView>
    <View>
      <PaddedView
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <UserAvatarUpdate />
        <H3>
          
        </H3>
      </PaddedView>
      <PaddedView />
      <TableView>
        <Touchable
          onPress={[Function]}
        >
          <Row
            accessoryComponent={
              <Unknown>
                New
              </Unknown>
            }
            accessoryText="1"
            leadingIcon="users-three"
            title="Following Me"
          />
        </Touchable>
        <Touchable
          onPress={[Function]}
        >
          <Row
            accessoryText="1"
            leadingIcon="user-circle"
            title="I'm Following"
          />
        </Touchable>
        <Touchable
          onPress={[Function]}
        >
          <Row
            accessoryText="0"
            leadingIcon="heart"
            title="Liked Content"
          />
        </Touchable>
      </TableView>
      <PaddedView />
      <TableView
        headerText="My Profile"
      >
        <Touchable
          onPress={[Function]}
        >
          <Row
            leadingIcon="user"
            title="Personal Details"
          />
        </Touchable>
        <Touchable
          onPress={[Function]}
        >
          <Row
            accessoryText="Select"
            leadingIcon="navigation-arrow"
            title="My Campus"
          />
        </Touchable>
      </TableView>
      <PaddedView />
      <TableView
        headerText="Settings"
      >
        <Touchable
          onPress={[Function]}
        >
          <Row
            leadingIcon="lock-simple"
            title="Change Password"
          />
        </Touchable>
        <Touchable
          onPress={[Function]}
        >
          <Row
            leadingIcon="bell"
            title="Notification Settings"
          />
        </Touchable>
      </TableView>
      <PaddedView />
      <TableView>
        <Touchable
          onPress={[Function]}
        >
          <Row
            leadingIcon="arrow-square-out"
            title="Logout"
          />
        </Touchable>
      </TableView>
      <H6
        style={[Function]}
      >
        App Version: 1.0.0.1
      </H6>
      <PaddedView />
      <TableView
        headerText="Developer Tools"
      >
        <Touchable
          onPress={[Function]}
        >
          <Row
            title="Launch OpenID Flow"
          />
        </Touchable>
        <Divider />
        <Touchable
          onPress={[Function]}
        >
          <Row
            title="Launch Onboarding"
          />
        </Touchable>
        <Divider />
        <Touchable
          onPress={[Function]}
        >
          <Row
            title="Reset Onboarding to Unseen"
          />
        </Touchable>
        <Divider />
        <Touchable
          onPress={[Function]}
        >
          <Row
            title="Reset Onboarding to Seen v1"
          />
        </Touchable>
      </TableView>
      <PaddedView />
    </View>
  </RCTScrollView>
</BackgroundView>
`;
